#! /bin/sh /usr/share/dpatch/dpatch-run
## 34-kfreebsd.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix the very same problem affecting linux on Debian GNU/kFreeBSD.
## DP: status: accepted upstream.

@DPATCH@

--- ACE_wrappers/TAO/TAO_IDL/be/be_codegen.cpp-
+++ ACE_wrappers/TAO/TAO_IDL/be/be_codegen.cpp
@@ -67,7 +67,7 @@
   delete this->server_inline_;
   delete this->server_template_inline_;
   delete this->anyop_source_;
-#if !defined (linux) && !defined (__QNX__)
+#if !defined (linux) && !defined (__QNX__) && !defined(__GLIBC__)
   // This causes a seg fault on Linux RH 5.1.  Let it leak . . .
   delete this->gperf_input_stream_;
 #endif /* ! linux */
@@ -1184,7 +1184,7 @@
 void
 TAO_CodeGen::gperf_input_stream (TAO_OutStream *os)
 {
-#if !defined (linux) && !defined (__QNX__)
+#if !defined (linux) && !defined (__QNX__) && !defined (__GLIBC__)
   // This causes a seg fault on Linux RH 5.1.  Let it leak . . .
   delete this->gperf_input_stream_;
 #endif /* ! linux */
